api/runtime/hooks/v1alpha1/upgrade_plan_types.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1beta1 "sigs.k8s.io/cluster-api/api/core/v1beta1"
+	runtimecatalog "sigs.k8s.io/cluster-api/exp/runtime/catalog"
+)
+
+// GenerateUpgradePlanRequest is the request of the GenerateUpgradePlan hook.
+// +kubebuilder:object:root=true
+type GenerateUpgradePlanRequest struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// CommonRequest contains fields common to all request types.
+	CommonRequest `json:",inline"`
+
+	// cluster is the cluster ofject the lifecycle hook correspods to.
+	// +required
+	Cluster clusterv1beta1.Cluster `json:"cluster,omitempty,omitzero"`
+
+	// fromControlPlaneKubernetesVersion is the current Kubernetes version of the control plane.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	FromControlPlaneKubernetesVersion string `json:"fromControlPlaneKubernetesVersion,omitempty"`
+
+	// fromWorkersKubernetesVersion is the current Kubernetes version of the workers.
```

Comment:
```suggestion
	// fromWorkersKubernetesVersion is the min Kubernetes version of the workers (MachineDeployments and MachinePools).
```

Reply:
Maybe let's say "min current", I think it's important to mention current